src/arbcall.jl

```diff
@@ -0,0 +1,76 @@
+
+function jlfname(arbfname,
+        prefixes=("arf", "arb", "acb", "mag"),
+        suffixes=("si", "ui", "d", "arf", "arb");
+        inplace=true)
+    strs = split(arbfname, "_")
+    k = findfirst(s->s ∉ prefixes, strs)
+    l = findfirst(s->s ∉ suffixes, reverse(strs))
+    fname = join(strs[k:end-l+1], "_")
+    return inplace ? Symbol(fname, "!") : Symbol(fname)
+end
+
+const Ctypes = Dict{String, DataType}(
+    "void"      => Cvoid,
+    "int"       => Cint,
+    "slong"     => Clong,
+    "ulong"     => Culong,
+    "double"    => Cdouble,
+    "arf_t"     => Arf,
+    "arb_t"     => Arb,
+    "acb_t"     => Acb,
+    "mag_t"     => Mag,
+    "arf_rnd_t" => arb_rnd
+)
+
+
+macro arbcall_str(str)
+    header_regex = r"(?<returntype>\w+)\s+(?<arbfunction>[\w_]+)\((?<args>.*)\)"
+
+    m = match(header_regex, str)
+
+    returnT = Ctypes[m[:returntype]]
+
+    args = match.(r"(?<const>const)?\s*(?<type>\w+)\s+(?<name>\w+)",
+        strip.(split(m[:args], ","))
+        )
+    arg_names = Symbol.([m[:name] for m in args])
+    jl_types = [Ctypes[m[:type]] for m in args]
+    c_types = [T ∈ (Arf, Arb, Acb, Mag) ? Ref{T} : T for T in jl_types]
+    jl_args = [:($a::$T) for (a, T) in zip(arg_names, jl_types)]
+
+    arbf = String(m[:arbfunction])
+    inplace = isnothing(args[1][:const]) && c_types[1] <: Ref
+    jlf = jlfname(arbf, inplace=true)
```

Comment:
```suggestion
    jlf = jlfname(arbf, inplace=inplace)
```

Reply:
although I think it would be better to make `jlf` an optional argument to the macro, for more flexibility